### ISSUE TITLE
tex_image_2d overload taking an HtmlImageElement isn't available in WebGL 2

### ISSUE THREAD

**Issue by nstoddard** [rustwasm/wasm-bindgen#1195]
I'm trying to use `tex_image_2d_with_u32_and_u32_and_image` in WebGL 2, but it seems to only be available in WebGL 1: https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebGlRenderingContext.html?search=tex_image_2d_with_u32_and_u32_and_image.

From [the IDL](https://github.com/rustwasm/wasm-bindgen/blob/master/crates/web-sys/webidls/enabled/WebGL2RenderingContext.webidl), it looks like WebGL 2 also adds another overload that takes `HtmlImageElement` plus some new parameters. Why isn't this overload available in `web-sys`?

**Comment by nstoddard** [comment]
Yep, WebGL 2 adds several more overloads for `texImage2d`. But all the overloads from version 1 are still available, and ideally those would have the same name in both versions. Looks like the only difference between the two versions is that the parameter name is different between [WebGL 1](https://github.com/rustwasm/wasm-bindgen/blob/master/crates/web-sys/webidls/enabled/WebGLRenderingContext.webidl#L752) and [2](https://github.com/rustwasm/wasm-bindgen/blob/master/crates/web-sys/webidls/enabled/WebGL2RenderingContext.webidl#L362), which could be the cause.

**Comment by alexcrichton** [comment]
We could try to update the WebIDL directly, but that unfortunately may be a breaking change :(

We sort of just do our best in terms of naming and it doesn't always turn out as well as one might expect! Otherwise though it's roughly intended that web-sys is generally how APIs are built rather than the end-point of usage
